search_params.py

```diff
@@ -235,8 +238,25 @@ def main():
         progress_reporter=reporter,
         config=config)
 
+    # Save best model after parameter search.
+    if args.save_best_model:
```

Comment:
we always save the best model after parameter selection. The flow should be like:

if train_val
  retrain
  save the model obtained from retraining
else
  save model